File: 1127-last-stone-weight/last-stone-weight.py
class Solution:
    def lastStoneWeight(self, stones: List[int]) -> int:
        heap=[-num for num in stones]
        heapq.heapify(heap)
        while len(heap)>1:
            a=heapq.heappop(heap)
            b=heapq.heappop(heap)
            temp=a-b
            heapq.heappush(heap,temp)
        return -heap[0]

File: 1127-last-stone-weight/last-stone-weight.java
class Solution {
    public int lastStoneWeight(int[] stones) {
        PriorityQueue<Integer> heap;
        heap = new PriorityQueue<>(Collections.reverseOrder());

        for (int num:stones){
            heap.add(num);
        }

        while (heap.size()>1) {
            heap.add(heap.poll()-heap.poll());
        }
        return heap.peek();
    }
}